### PR TITLE
Enforce logging when killing a task

### PR DESCRIPTION
**Pull request by jihoonson**
Logging when a task is killed is important because sometimes it's hard to imagine why it was killed, especially when some process killed rather than human like in kafka indexing service.

### CODE REVIEW

File: extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisor.java [hunk 1]
```diff
@@ -839,8 +839,7 @@ private void killTasksInGroup(TaskGroup taskGroup)
   {
     if (taskGroup != null) {
       for (String taskId : taskGroup.tasks.keySet()) {
-        log.info("Killing task [%s] in the task group", taskId);
-        killTask(taskId);
+        killTask(taskId, "Killing task [%s] in the task group[%s]", taskId, taskGroup.groupId);
```
Comment (QiuMM) [inline extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisor.java]: I think add a reason to explain why killing a task in a task group would be better. Maybe modify `private void killTasksInGroup(TaskGroup taskGroup)` to `private void killTasksInGroup(TaskGroup taskGroup, String reason)`.

Comment (jihoonson) [reply extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisor.java]: Good point. Fixed. Thanks!

File: extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java [hunk 2]
```diff
@@ -346,7 +346,8 @@ void checkSegmentsAndSubmitTasks()
           && !toBuildInterval.get(interval).equals(runningVersion.get(interval))
       ) {
         if (taskMaster.getTaskQueue().isPresent()) {
-          taskMaster.getTaskQueue().get().shutdown(runningTasks.get(interval).getId());
+          taskMaster.getTaskQueue().get().shutdown(
+              runningTasks.get(interval).getId(), "version mismatch");
```
Comment (asdf2014) [inline extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java]: There is no need to break this down into a new line.

Comment (jihoonson) [reply extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java]: Thanks, fixed.

**Pull request merged**